apps/greenhouse/src/components/AuthProvider.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { createContext, useContext, useState, useMemo, useCallback, useRef } from "react"
+import { oidcSession, mockedSession } from "@cloudoperators/juno-oauth"
+
+const setOrganizationToUrl = (groups) => {
+  const orgString = groups?.find((g) => g.indexOf("organization:") === 0)
+  if (orgString) {
+    const name = orgString.split(":")[1]
+    let url = new URL(window.location.href)
+    url.searchParams.set("org", name)
+    window.history.replaceState(null, null, url.href)
+  }
+}
+
+const AuthContext = createContext()
+
+export const AuthProvider = ({ options, children }) => {
+  const [authData, setAuthData] = useState(null)
+  const [isDemoMode, setIsDemoMode] = useState(false)
+  const oidcRef = useRef(null)
+
+  const initializeOidc = useCallback(() => {
+    if (oidcRef.current) return oidcRef.current
+    setIsDemoMode(false)
+
+    // extract auth props
+    const issuerURL = options?.authIssuerUrl
```

Comment:
Non-blocking change!
This could be improved by using destructuring.

Reply:
done! thanks